packages/content/lib/content-sources/sanity-source.js

```diff
@@ -0,0 +1,273 @@
+import { toHTML } from '@portabletext/to-html';
+import toMarkdown from '@sanity/block-content-to-markdown';
+
+/**
+ * Options for SanitySource
+ */
+export class SanityOptions extends SourceOptions {
+	constructor({
+		apiVersion = "v2021-10-21",
+		projectid = undefined,
+		dataset = "production",
+		apiToken = undefined,
+		useCdn = false,
+    baseUrl = undefined,
+		queries = [],
+		textConverters = [],
+		limit = 100,
+		maxNumPages = -1,
+		pageNumZeroPad = 0,
+		...rest
+	} = {}) {
+
+    super(rest);
+
+		/**
+		 * API Version
+		 * @type {string}
+		 */
+		this.apiVersion = apiVersion;
+
+		/**
+		 * Sanity Project ID
+		 * @type {string}
+		 */
+		this.projectid = projectid;
+
+		/**
+		 * API Version
+		 * @type {string}
+		 */
+		this.dataset = dataset;
+
+		/**
+		 * `false` if you want to ensure fresh data
+		 * @type {string}
+		 */
+		this.useCdn = useCdn;
+
+		/**
+		 * The base url of your Sanity CMS (with or without trailing slash).
+		 * @type {string}
+		 */
+		this.baseUrl = baseUrl;
+
+		/**
+		 *
+		 * @type {Array.<string>}
+		 */
+		this.queries = queries;
+
+		/**
+		 *
+		 * @type {Array.<string>}
+		 */
+		this.textConverters = textConverters;
+
+		/**
+		 * Max number of entries per page. Default is 100.
+		 * @type {number}
+		 */
+		this.limit = limit;
+
+		/**
+		 * Max number of pages. Default is -1 for all pages
+		 * @type {number}
+		 */
+		this.maxNumPages = maxNumPages;
+
+		/**
+		 * How many zeros to pad each json filename index with. Default is 0
+		 * @type {number}
+		 */
+		this.pageNumZeroPad = pageNumZeroPad;
+
+		/**
+		 * API Token defined in your sanity project.
+		 * @type {string}
+		 */
+    this.apiToken = apiToken;
+  }
+}
+
+class SanitySource extends ContentSource {
+
+	/**
+	 *
+	 * @param {*} config
+	 * @param {Logger} logger
+	 */
+	constructor(config, logger) {
+		super(SanitySource._assembleConfig(config), logger);
+
+		// Build and init the sanity Client to use here.
+		this.client = sanityClient({
+			projectId: this.config.projectid,
+			dataset: this.config.dataset,
+			apiVersion: this.config.apiVersion, // use current UTC date - see "specifying API version"!
+			token: this.config.apiToken, // or leave blank for unauthenticated usage
+			useCdn: this.config.useCdn, // `false` if you want to ensure fresh data
+		});
+  }
+
+  /**
+   * @returns {Promise<ContentResult>}
+   */
+  async fetchContent() {
+		const result = new ContentResult();
+
+		for (const query of this.config.queries) {
+			await this._fetchPages(query, result, {
+				start: 0,
+				limit: this.config.limit
+			});
+		}
+
+		return result;
+  }
+
+  /**
+   * Recursively fetches content using the Sanity client.
+   *
+   * @param {string} query
+	 * @param {string} jwt The JSON web token generated by Sanity
+   * @param {ContentResult} result
+   * @param {Object} params
+   * @returns {Promise<Object>} Object with an 'entries' and an 'assets' array.
+   */
+  async _fetchPages(
```

Comment:
I haven't looked much into Sanity's SDK, but how close is it to a general GraphQL API? Or is it using Sanity's own GROQ?

Reply:
Its GROQ. I think we could have this setup as just generic grab data... but then also add custom queries.